common/persistence/sql/sqlTaskPersistence.go

```diff
@@ -51,21 +115,243 @@ func (m *sqlTaskManager) Close() {
+				}
+			}
+		}
+
+		if err := tx.Commit(); err != nil {
+			return nil, &workflow.InternalServiceError{
+				Message: fmt.Sprintf("LeaseTaskList operation failed. Failed to commit transaction. Error: %v", err),
+			}
+		}
+
+	}
+
+	return &persistence.LeaseTaskListResponse{&persistence.TaskListInfo{
+		request.DomainID,
+		request.TaskList,
+		request.TaskType,
+		rangeID + 1,
+		ackLevel,
+		request.TaskListKind,
+	}}, nil
 }
 
 func (m *sqlTaskManager) UpdateTaskList(request *persistence.UpdateTaskListRequest) (*persistence.UpdateTaskListResponse, error) {
-	panic("implement me")
+	tx, err := m.db.Beginx();
+	if err != nil {
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("UpdateTaskList operation failed. Failed to begin transaction. Error: %v", err),
+		}
+	}
+	defer tx.Rollback()
+
+	if err := lockAndCheckTaskListRangeID(tx, request.TaskListInfo.DomainID, request.TaskListInfo.Name, request.TaskListInfo.TaskType, request.TaskListInfo.RangeID); err != nil {
+		switch err.(type) {
+		case *persistence.ConditionFailedError:
+			return nil, err
+		default:
+			return nil, &workflow.InternalServiceError{
+				Message: fmt.Sprintf("UpdateTaskList operation failed. Error: %v", err),
+			}
+		}
+	}
+
+	if result, err := tx.NamedExec(updateTaskListSQLQuery,
+		struct {
+		tasksListsRow
+		OldRangeID int64 `db:"old_range_id"`
+	}{
+		tasksListsRow{
+			request.TaskListInfo.DomainID,
+			request.TaskListInfo.RangeID,
+			request.TaskListInfo.Name,
+			int64(request.TaskListInfo.TaskType),
+			request.TaskListInfo.AckLevel,
+			int64(request.TaskListInfo.Kind),
+			MaximumExpiryTs,
+		},
+		request.TaskListInfo.RangeID,
+	}); err != nil {
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("UpdateTaskList operation failed. Failed to update task list. Error: %v", err),
+		}
+	} else {
+		rowsAffected, err := result.RowsAffected()
+		if err != nil {
+			return nil, &workflow.InternalServiceError{
+				Message: fmt.Sprintf("UpdateTaskList operation failed. Failed to verify how many rows were affected. Error: %v", err),
+			}
+		}
+		if rowsAffected != 1 {
+			return nil, &workflow.InternalServiceError{
+				Message: fmt.Sprintf("UpdateTaskList operation failed. %v rows were affected instead of 1.", rowsAffected),
+			}
+		}
+	}
+
+	if err := tx.Commit(); err != nil {
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("UpdateTaskList operation failed. Failed to commit transaction. Error: %v", err),
+		}
+	}
+
+	return &persistence.UpdateTaskListResponse{}, nil
 }
 
 func (m *sqlTaskManager) CreateTasks(request *persistence.CreateTasksRequest) (*persistence.CreateTasksResponse, error) {
-	return nil, nil
+	tx, err := m.db.Beginx()
+	if err != nil {
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("CreateTasks operation failed. Failed to begin transaction. Error: %v", err),
+		}
+	}
+	defer tx.Rollback()
+
+	structs := make([]tasksRow, len(request.Tasks))
```

Comment:
General comment: make variable names more meaningful. for example structs -> tasks or taskRows